src/users/containers/user-details/user-details.component.ts

```diff
@@ -19,28 +18,85 @@ export class UserDetailsComponent implements OnInit, OnDestroy {
 
   userSubscription: Subscription;
   dependanciesSubscription: Subscription;
+  suspendUserServerErrorSubscription: Subscription;
+
+  actionButtons: { name: string, class: string, action: () => {} }[] = [];
+
+  suspendViewFlag: boolean = false;
+
+  showSuspendView: () => {};
+  hideSuspendView: () => {};
+
+  suspendSuccessSubscription: Subscription;
 
   constructor(
     private userStore: Store<fromStore.UserState>,
     private routerStore: Store<fromRoot.State>,
+    private actions$: Actions,
   ) { }
 
   ngOnInit(): void {
+    this.user$ = new Observable();
+    this.setSuspendViewFunctions();
+
     this.isLoading$ = this.userStore.pipe(select(fromStore.getGetUserLoading));
 
-    this.dependanciesSubscription = combineLatest([
-      this.routerStore.pipe(select(fromRoot.getRouterState)),
-      this.userStore.pipe(select(fromStore.getGetUserLoaded))
-    ]).subscribe(([route, users]) => {
-      if (users === false) {
-        this.userStore.dispatch(new fromStore.LoadUsers());
-      }
-      const userId = route.state.params.userId;
-      this.user$ = this.userStore.pipe(select(fromStore.getGetSingleUser, { userIdentifier: userId }));
+    this.dependanciesSubscription = this.getDependancyObservables(this.userStore, this.routerStore).subscribe(([route, users]) => {
+      this.handleDependanciesSubscription(users, route);
+    });
+
+    this.userSubscription = this.user$.subscribe((user) => this.handleUserSubscription(user));
+
+    this.suspendSuccessSubscription = this.actions$.pipe(ofType(fromStore.SUSPEND_USER_SUCCESS)).subscribe(() => {
```

Comment:
move this to effects... this all component is going against all the rules we have